spec/features/etd_creation_spec.rb

```diff
@@ -216,40 +216,32 @@
     status_element = find_table_cell_following(header_text: 'Status')
     expect(status_element).to have_text('v1 Registered')
-    if Settings.folio.enabled
-      reload_page_until_timeout! do
+
+    Timeout.timeout(Settings.timeouts.workflow) do
+      workflow_done = false
+      loop do
         click_link('etdSubmitWF')
-        # expect first 6 steps to have completed w/ Folio (but requires reload since the Folio job takes time)
-        page.has_text?(/register-object completed/) &&
-          page.has_text?(/submit completed/) &&
-          page.has_text?(/reader-approval completed/) &&
-          page.has_text?(/registrar-approval completed/) &&
-          page.has_text?(/submit-marc completed/) &&
-          page.has_text?(/check-marc completed/)
-
-        # NOTE: it would be lovely if we could process the ETD through the rest of the etdSubmitWF steps
-        #   and then run it through common-accessioning, but the remaining etdSubmitWF steps, catalog-status and
-        #   otherMetadata, require too much fakery specific to the ETD app (cron job, cocina-model updates)
-        #   to make sense here.
-      end
-    else
-      click_link('etdSubmitWF')
-      within('#blacklight-modal') do
-        # expect first 5 steps to have completed
-        expect(page).to have_text(/register-object completed/)
-        expect(page).to have_text(/submit completed/)
-        expect(page).to have_text(/reader-approval completed/)
-        expect(page).to have_text(/registrar-approval completed/)
-        expect(page).to have_text(/submit-marc completed/)
-        expect(page).to have_text(/check-marc waiting/)
-
-        # NOTE: it would be lovely if we could process the ETD through the rest of the etdSubmitWF steps
-        #   and then run it through common-accessioning, but the remaining etdSubmitWF steps of check-marc,
-        #   catalog-status, and otherMetadata require too much fakery specific to the ETD app (cocina-model
-        #   updates) to make sense here.
+        within('.modal-dialog') do
+          workflow_done = true if page.has_text?(/register-object\s+completed/) &&
+            page.has_text?(/register-object\s+completed/) &&
+            page.has_text?(/submit\s+completed/) &&
+            page.has_text?(/reader-approval\s+completed/) &&
+            page.has_text?(/registrar-approval\s+completed/) &&
+            page.has_text?(/submit-marc\s+completed/, wait: 1) &&
+            page.has_text?(/check-marc\s+completed/, wait: 1) &&
+            page.has_text?(/catalog-status\s+waiting/, wait: 1)
+
+          # NOTE: it would be lovely if we could process the ETD through the rest of the etdSubmitWF steps
+          #   and then run it through common-accessioning, but the remaining etdSubmitWF steps, catalog-status and
+          #   otherMetadata, require too much fakery specific to the ETD app (cron job, cocina-model updates)
+          #   to make sense here.
+
+          page.send_keys(:escape)
```

Comment:
it turns out that "Cancel" is not unique on the page and we don't have identifiers or other unique selectors on these blacklight modal buttons.  So I opted to go with the escape key to close the modal.